rules/hmap.bzl

```diff
@@ -51,16 +23,17 @@ def _make_headermap_impl(ctx):
         else:
             fail("direct_hdr_provider %s must contain either 'CcInfo' or 'objc' provider" % provider)
 
-    out = _make_headermap_input_file(ctx.attr.namespace, all_hdrs, ctx.attr.flatten_headers)
     ctx.actions.write(
```

Comment:
instead of manually writing this file, could we use `ctx.action.args()` to create a params file, so we don't ever need to materialize all the headers to a list during analysis?

Reply:
I started going down this path but there are more changes required since the `hmapbuild` tool doesn't really support param files. Once the cleanups in #61 are in I can add param file support and turn this on.

Reply:
i meant just cheating to use the param file to write out the list, instead of using it as a legit param file. Feel free to land as-is and I can try that out

Reply:
I tried that first but it’s not that simple. You can’t really control _when_ Bazel will stop adding args and use a param file. You can set it to _always_ use a param file though.